tests/components/template/test_sensor.py

```diff
@@ -376,6 +377,52 @@ def test_setup_valid_device_class(self):
         state = self.hass.states.get("sensor.test2")
         assert "device_class" not in state.attributes
 
+    def test_available_template_with_entities(self):
+        """Test availability tempalates with values from other entities."""
+        availability_template = """
+            {% if is_state('availability_boolean.state', 'True') %}
+                {{ 'true' }}
+            {% else %}
+                {{ 'false' }}
+            {% endif %}
+        """
+
+        with assert_setup_component(1):
+            assert setup_component(
+                self.hass,
+                "sensor",
+                {
+                    "sensor": {
+                        "platform": "template",
+                        "sensors": {
+                            "test_template_sensor": {
+                                "value_template": "{{ states.sensor.test_state.state }}",
+                                "availability_template": availability_template,
+                            }
+                        },
+                    }
+                },
+            )
+
+        self.hass.start()
+        self.hass.block_till_done()
+
+        # When template returns true..
+        self.hass.states.set("availability_boolean.state", True)
```

Comment:
We only allow string values for states, please use `"on"` and `"off"`

Reply:
Done